server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdMonitor.java

```diff
@@ -373,6 +374,15 @@ public void onNewInfo(ClusterInfo info) {
         } else {
             listener.onResponse(null);
         }
+
+        // If all the nodes are breaching high disk watermark, we apply index create block to avoid red clusters.
+        if (nodesOverHighThreshold.size() == nodes.size()) {
+            applyIndexCreateBlock(listener, true);
+        } else if (state.getBlocks().hasGlobalBlockWithId(INDEX_CREATE_BLOCK_ID)) {
+            applyIndexCreateBlock(listener, false);
```

Comment:
I think we can still see RED index if the Index setting excludes the nodes from shard allocation due to shard allocation filter and only those nodes have not breached the High disk watermark settings.
Please add that in comments to mention this limitation.